staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -952,7 +952,7 @@ func mergeKeyValueEqual(left, right interface{}, mergeKey string) (bool, error)
 	if !ok {
 		return false, mergepatch.ErrNoMergeKey(typedRight, mergeKey)
 	}
-	return mergeKeyLeft == mergeKeyRight, nil
+	return reflect.DeepEqual(mergeKeyLeft, mergeKeyRight), nil
```

Comment:
Do we need to add test cases to cover this change? As I see it, mergeKeyLeft == mergeKeyRight just checks if the two interface values are equal, while reflect.DeepEqual(mergeKeyLeft, mergeKeyRight) goes deeper and compares the actual content of the values. They're not the same.

Reply:
Yeah, they are not same. To me, the purpose of this function is trying to compare two values/contents are equal or not. with `==`, it will raise panic when two values are map. So I think there is no case for this function just check the reference not the content. So I think deepEqual will cover the more cases than `==` especially in panic case or we want to return error in this case? Please correct me if I am wrong.

Reply:
Yes, you are right! I mean, because the original one has a bug, right? Then do we have unit tests to cover this bug? I see that this method originally has no test cases, do we need to add them to cover the content of this change?

Reply:
we do need unit tests, please unresolve this comment